packages/aws-cdk-lib/aws-ec2/lib/nat.ts

```diff
@@ -399,6 +419,115 @@ class PrefSet<A> {
   }
 }
 
+/**
+ * Modern NAT provider which uses NAT Instances.
+ * The instance uses Amazon Linux 2 as the operating system.
+ */
+export class NatInstanceProviderAL2 extends NatProvider implements IConnectable {
+  private gateways: PrefSet<Instance> = new PrefSet<Instance>();
+  private _securityGroup?: ISecurityGroup;
+  private _connections?: Connections;
+
+  constructor(private readonly props: NatInstanceProps) {
+    super();
+
+    if (props.defaultAllowedTraffic !== undefined && props.allowAllTraffic !== undefined) {
+      throw new Error('Can not specify both of \'defaultAllowedTraffic\' and \'defaultAllowedTraffic\'; prefer \'defaultAllowedTraffic\'');
+    }
+
+    if (props.keyName && props.keyPair) {
+      throw new Error('Cannot specify both of \'keyName\' and \'keyPair\'; prefer \'keyPair\'');
+    }
+  }
+
+  public configureNat(options: ConfigureNatOptions) {
+    const defaultDirection = this.props.defaultAllowedTraffic ??
+      (this.props.allowAllTraffic ?? true ? NatTrafficDirection.INBOUND_AND_OUTBOUND : NatTrafficDirection.OUTBOUND_ONLY);
+
+    // Create the NAT instances. They can share a security group and a Role.
+    const machineImage = this.props.machineImage || new AmazonLinuxImage({ generation: AmazonLinuxGeneration.AMAZON_LINUX_2 });
```

Comment:
Is there any reason not to use AL2023 instead of AL2 by default? AL2 will [reach EOL](https://aws.amazon.com/amazon-linux-2/faqs/) next year, whereas AL2023 [will be supported](https://aws.amazon.com/linux/amazon-linux-2023/faqs/) until 2028.

Reply:
Good point, I'll update it to AL2023